lua/muren.lua

```diff
@@ -1,17 +1,54 @@
 local M = {}
 
+M.anchor_positions = {
+  'top',
+  'bottom',
+  'left',
+  'right',
+  'top_left',
+  'top_right',
+  'bottom_left',
+  'bottom_right',
+}
 
 M.setup = function(opts)
   local options = require('muren.options')
   options.update(opts or {})
 
+  local api = require('muren.api')
+  local muren_commands = {
+    MurenOpen = { api.open_ui, true },
+    MurenClose = { api.close_ui, nil },
+    MurenToggle = { api.toggle_ui, true },
+    MurenFresh = { api.open_fresh_ui, true },
+    MurenUnique = { api.open_unique_ui, true },
+  }
+
+
+  local create_muren_commands = function()
+    for name, muren_cmd in pairs(muren_commands) do
+      local muren_api, range = unpack(muren_cmd)
+      vim.api.nvim_create_user_command(name, function(args)
+        local anchor, vertical_offset, horizontal_offset = unpack(args.fargs)
+        muren_api({
+          anchor = anchor,
```

Comment:
```suggestion
          range = args.range,
          anchor = anchor,
```
such that substitution only in a visual selection works

Reply:
Thanks, sorry I overlooked that! Too hasty on the refactor.

Reply:
I made some fixes but I want to sleep on them and get a fresh look tomorrow! Main thing is that for the `get_range` function we also need line1 and line2 from the `args` here.
```suggestion
          range = args.range,
          line1 = args.line1,
          line2 = args.line2,
          anchor = anchor,
```

I'm also trying to think of some good-looking way to let a direct call to the api get the current visual selection, since e.g. `require("muren.api").open_ui({some_opts_here})` won't have the visual selection. My current hack and slash approach is to modify get_range like this:

```lua
local get_range = function(opts)
  local range
  if opts.range == 2 or vim.api.nvim_get_mode().mode == 'V' then
    local start_line, end_line = vim.fn.getpos("v")[2], vim.fn.getpos(".")[2]
    range = {start = opts.line1 or start_line, _end = opts.line2 or end_line}
  end
  return range
end
```

But I'll work on cleaning that up